Sources/xcodeproj/XcodeProj.swift

```diff
@@ -27,33 +25,68 @@ public struct XcodeProj {
         if pbxprojPaths.count == 0 {
             throw XCodeProjError.pbxprojNotFound(path: path)
         }
-        pbxproj = try PBXProj(path: pbxprojPaths.first!,
-                              name: path.lastComponentWithoutExtension)
+        pbxproj = try PBXProj(path: pbxprojPaths.first!)
         let xcworkspacePaths = path.glob("*.xcworkspace")
         if xcworkspacePaths.count == 0 {
             throw XCodeProjError.xcworkspaceNotFound(path: path)
         }
         workspace = try XCWorkspace(path: xcworkspacePaths.first!)
-        self.path = path
         let sharedDataPath = path + Path("xcshareddata")
         self.sharedData = try? XCSharedData(path: sharedDataPath)
     }
     
     /// Initializes the XCodeProj
     ///
     /// - Parameters:
-    ///   - path: project path
     ///   - workspace: project internal workspace.
     ///   - pbxproj: project .pbxproj.
-    public init(path: Path, workspace: XCWorkspace, pbxproj: PBXProj, sharedData: XCSharedData? = nil) {
-        self.path = path
+    public init(workspace: XCWorkspace, pbxproj: PBXProj, sharedData: XCSharedData? = nil) {
         self.workspace = workspace
         self.pbxproj = pbxproj
         self.sharedData = sharedData
     }
     
 }
 
+// MARK: - <Writable>
+
+extension XcodeProj: Writable {
+
+    public func write(path: Path, override: Bool = true) throws {
+        try path.mkpath()
+
+        // write workspace
+        let workspacePath = path + "project.xcworkspace"
+        try workspace.write(path: workspacePath, override: override)
+
+        // write pbxproj
+        let pbxprojPath = path + "project.pbxproj"
+        try pbxproj.write(path: pbxprojPath, override: override)
+
+        // write shared data
+        if let sharedData = sharedData {
+            let schemesPath = path + "xcshareddata/xcschemes"
+            try schemesPath.mkpath()
+            for scheme in sharedData.schemes {
+                try scheme.write(path: schemesPath + scheme.name, override: override)
+            }
+        }
+    }
+
+}
+
+// MARK: - XcodeProj Extension (Equatable)
+
+extension XcodeProj: Equatable {
+
+    public static func == (lhs: XcodeProj, rhs: XcodeProj) -> Bool {
+        return lhs.workspace == rhs.workspace &&
+            lhs.pbxproj == rhs.pbxproj
+            //TODO: make SharedData equatable: lhs.sharedData == rhs.sharedData
```

Comment:
I'd do this on a different PR, also conforming the protocol from `XCSharedData`.